### ISSUE TITLE
Warn if a variable is assigned by the result of a post increment/decrement expression

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#359]
A post increment or decrement expression first gets the current value and then does the increment or decrement without affecting the previously got current value.

So if the value is assigned to a different variable, then both variables always differ. This might be surprising and unwanted, so we should warn in such situation.

Examples:
```c#
var y = x--;
var z = x++;
```

Same is valid if placed inside parenthesis:
```c#
var y = (x--);
var z = (x++);
```

In the examples above, `y` and `z` always differ from `x` by `1`.

**Comment by RalfKoban** [comment]
Also consider method calls:

```c#
DoSomething(x--);
DoSomething(x++);
```
